web3/subgraph/src/crowdfund.ts

```diff
@@ -0,0 +1,112 @@
+import { BigInt, Address } from "@graphprotocol/graph-ts";
+import { Crowdfund, User, Donor } from "../generated/schema";
+import { donated, funded, refundActive, userBalanceWithdrawn } from '../generated/templates/Crowdfund/Crowdfund';
+
+export function donationHandler(event: donated): void {
+  let crowdfundContractAddress = event.address.toHexString();
+  let donorAddress = event.params.sender.toHexString();
+  let amount = event.params.amount;
+
+  let crowdfund = Crowdfund.load(crowdfundContractAddress);
+
+  if (crowdfund === null) {
+    // this shouldn't be indexed given that it was never instantiated
+    return;
+  }
+
+  let user = User.load(donorAddress);
+
+  if (user === null) {
+    user = new User(donorAddress);
+    user.address = event.params.sender;
+    user.createdAt = event.block.timestamp;
+  }
+
+  let fundDonor = Donor.load(`${crowdfund.id}-${user.id}`)
+
+  if (fundDonor === null) {
+    fundDonor = new Donor(`${crowdfund.id}-${user.id}`)
+    fundDonor.amount = amount;
+    fundDonor.balance = amount;
+    fundDonor.crowdfund = crowdfund.id;
+    fundDonor.createdAt = event.block.timestamp;
+  } else {
+    // increment balance & amount
+    fundDonor.amount = fundDonor.amount.plus(amount);
+    fundDonor.balance = fundDonor.balance.plus(amount);
```

Comment:
Is there any difference between amount and balance?

Reply:
Amount is used to track totally amount donated by user even if the goal isn't reached and withdrawals are enabled